seonshine_frontend/src/pages/signUp/ProfileRegistration/index.tsx

```diff
@@ -0,0 +1,260 @@
+import { zodResolver } from '@hookform/resolvers/zod';
+import { Visibility, VisibilityOff } from '@mui/icons-material';
+import { Box, Button, FormControl, IconButton, InputLabel, MenuItem, Select, Stack, Typography } from '@mui/material';
+
+import FormInput from '@/components/molecules/formEntity/input';
+import { FormLabel } from '@/components/molecules/formEntity/label';
+
+import { useSignUpApi } from '@/apis/hooks/signUpApi.hook';
+import { useGetBranches } from '@/apis/hooks/userApi.hook';
+
+import { EnterUserInformationPropsType } from '../types';
+import { SignUpSchema, SignUpSchemaType } from './schema';
+
+const ProfileRegistration = ({
+  handleSubmitInformation,
+  userType,
+}: EnterUserInformationPropsType & { userType: string }) => {
+  const {
+    handleSubmit,
+    register,
+    setValue,
+    formState: { errors },
+  } = useForm<SignUpSchemaType>({
+    resolver: zodResolver(SignUpSchema),
+    defaultValues: {
+      userType,
+      employeeId: '',
+      password: '',
+      confirmPassword: '',
+      fullName: '',
+      email: '',
+      phoneNumber: '',
+      branch_id: 0,
+    },
+  });
+  const { data: branchData } = useGetBranches({ enabled: true });
+
+  const submitForm = (data: SignUpSchemaType) => {
+    handleSubmitInformation({ ...data, userType, branch_id: selectedBranch });
+  };
+  // will use custom hook useSignUp after having the API
+  // const { mutate: signUpUser } = useSignUpApi();
+
+  const [showPassword, setShowPassword] = useState<boolean>(false);
+  const [showConfirmPassword, setShowConfirmPassword] = useState<boolean>(false);
+  const handleClickShowPassword = () => setShowPassword((prev) => !prev);
+  const handleClickShowConfirmPassword = () => setShowConfirmPassword((prev) => !prev);
+  const [selectedBranch, setSelectedBranch] = useState('');
+
+  return (
+    <Stack
+      alignItems="center"
+      justifyContent="center"
+      className="min-h-[80vh]"
+    >
+      <Stack
+        direction="column"
+        alignItems="center"
+        className="w-full px-24 bg-white rounded-lg shadow-md sm:w-150 md:w-150 xl:w-150 max-w-screen"
+      >
+        <Box className="grid w-full h-full grid-cols-1 lg:grid-cols-1">
+          <Typography
+            variant="heading2"
+            component="h2"
+            className="text-center pt-8"
+          >
+            Sign Up
+          </Typography>
+          <Stack
+            justifyContent="center"
+            alignItems="center"
+            className="h-full px-2"
+          >
+            <form
+              onSubmit={handleSubmit(submitForm)}
+              className="w-full"
+              noValidate
+            >
+              <Stack
+                flexDirection="column"
+                className="h-full"
+              >
+                <Box className="grid gap-2 mt-4">
+                  <Stack direction="column">
+                    <FormLabel
+                      title={userType == '1' ? 'Employee Number' : 'ID'}
```

Comment:
With user type. we shouldn't hardcode like this, it was hard to understand. Please define UserRoleEnum if not exist. Please update for all
We can define constant mapping. Ex `labelByRole: {[Role.Restaurant]: 'ID'}`